### ISSUE TITLE
`@stylistic/number-leading-zero` and `@stylistic/number-no-trailing-zeros` drop every comment a value or a set of parameters holds

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#268]
Found while fixing [#233](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/233), whose parting claim — that `unit-case` was the last rule of the plugin writing that way — is wrong. Two more rules write `node.value` and `node.params` directly, and lose the same comments in the same silence.

### What minimal example or steps are needed to reproduce the bug?

```css
a { b: 1px /* c */ 0.50em; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/number-leading-zero": "never"
  }
}
```

`@stylistic/number-no-trailing-zeros` set to `true` loses the same comment on the same file.

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

`a { b: 1px /* c */ .50em; }` — the zero taken away, and every other character of the declaration left exactly as it was.

### What actually happened?

```css
a { b: 1px  .50em; }
```

The comment is gone, and nothing is reported. A set of at-rule parameters goes the same way, and so does the value of a Less at-variable. Under `postcss-scss` a comment written with a double slash is lost too: `a { b: 1px // c` with `0.50em; }` on the line under it comes back with the comment gone and the two lines still apart.

### What is the cause?

The one [#233](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/233) and [#235](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/235) name. PostCSS keeps the comments of a value in `raws.value.raw` and hands `decl.value` back cleaned of them, and `postcss-scss` keeps a third copy in `raws.value.scss`, which is the one it prints. Writing the bare property discards both raws, so the value the file gets is the cleaned one.

Both rules read `decl.value` and `atRule.params` and write to the same bare properties:

```js
root.walkDecls((decl) => check(decl, decl.value))
…
node.params = removeLeadingZeros(node.params, startIndex, endIndex)
syncLessVariableValue(node, node.params)
…
node.value = removeLeadingZeros(node.value, startIndex, endIndex)
```

The corpus of the comment oracle never caught these two because no fixture of it puts a number carrying a zero beside a comment; a fixture that does finds them at once.

### Do you have a proposal to fix the bug?

The same move [#233](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/233) makes: read through `getDeclarationValue` and `getAtRuleParams`, write through `setDeclarationValue` and `setAtRuleParams`. The setter for an at-rule already syncs the value of a Less at-variable, so the explicit call goes with it. Both rules write at an index rather than printing a parse back, and the index is measured in the text the getter hands over, which is the text the positions of the warnings are already counted in.
